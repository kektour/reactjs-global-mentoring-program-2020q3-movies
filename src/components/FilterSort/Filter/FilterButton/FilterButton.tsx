import React, { useCallback } from 'react';
import classnames from 'classnames';
import { Item } from '../Filter';
import styles from './FilterButton.module.scss';

type Props = {
  item: Item;
  isSelected: boolean;
  onClick: (id: string) => void;
};

export const FilterButton: React.FC<Props> = (props) => {
  const { item, isSelected, onClick } = props;
  const handleClick = useCallback(() => onClick(item.id), [item, onClick]);
  return (
    <div className={styles.root}>
      <button
        className={classnames({
          [styles.button]: true,
          [styles.buttonActive]: isSelected,
        })}
        onClick={handleClick}
      >
        {item.name}
      </button>
      {isSelected && <div className={styles.line}></div>}
    </div>
  );
};
