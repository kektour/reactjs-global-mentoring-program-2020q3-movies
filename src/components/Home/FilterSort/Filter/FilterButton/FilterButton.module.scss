@import 'src/shared.scss';

.root {
  margin-right: 30px;
  position: relative;

  &:last-child {
    margin-right: 0;
  }
}

.button {
  background: transparent;
  border: none;
  text-transform: uppercase;
  color: #fff;
  text-decoration: none;
  font-size: 16px;

  &:hover {
    cursor: pointer;
  }

  &:focus {
    outline: none;
  }
}

.buttonActive {
  color: #f65261;
}

.line {
  @extend %line;
  position: absolute;
  top: 42px;
  height: 2px;
  background-color: #f65261;
}
