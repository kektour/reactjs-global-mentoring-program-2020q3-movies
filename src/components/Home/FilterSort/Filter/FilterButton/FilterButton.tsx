import React, { useCallback } from 'react';
import classnames from 'classnames';
import styles from './FilterButton.module.scss';

type Props = {
  selected: boolean;
  onClick: () => void;
};

export const FilterButton: React.FC<Props> = (props) => {
  const { children, selected, onClick: handleClick } = props;
  const handleHrefClick = useCallback((e: React.MouseEvent<HTMLAnchorElement>) => {
    e.preventDefault();
    handleClick();
  }, [handleClick]);
  return (
    <div className={styles.root}>
      <a
        href="/"
        className={classnames({
          [styles.button]: true,
          [styles.buttonActive]: selected,
        })}
        onClick={handleHrefClick}
      >
        {children}
      </a>
      {selected && <div className={styles.line}></div>}
    </div>
  );
};
