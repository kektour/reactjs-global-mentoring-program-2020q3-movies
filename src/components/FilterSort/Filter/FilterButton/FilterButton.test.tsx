import React from 'react';
import { act, render, wait } from '@testing-library/react';
import { FilterButton } from './FilterButton';
import userEvent from '@testing-library/user-event';

describe('FilterButton', () => {
  it('Should render component correctly', () => {
    const { asFragment } = render(
      <FilterButton
        item={{
          id: 'foo',
          name: 'Foo',
        }}
        isSelected={false}
        onClick={() => null}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render component correctly', async () => {
    const handleClickMock = jest.fn();
    const { container } = render(
      <FilterButton
        item={{
          id: 'foo',
          name: 'Foo',
        }}
        isSelected={false}
        onClick={handleClickMock}
      />
    );
    act(() => userEvent.click(container.querySelector('button') as Element));
    await wait(() => {
      expect(handleClickMock).toHaveBeenCalledWith('foo');
    });
  });
});
