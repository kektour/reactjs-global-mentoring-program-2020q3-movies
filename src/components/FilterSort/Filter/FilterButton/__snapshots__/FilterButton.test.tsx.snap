// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton Should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <button
      class="button"
    >
      Foo
    </button>
  </div>
</DocumentFragment>
`;
